### PR TITLE
Upgrade to 0.37.0 cadl compiler and cadl ranch to the latest

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1649

In this pr we do following changes:

- Upgrade cadl compiler & cadl ranch to the latest
- Enable special words test cases in cadl-ranch
- Fix some issues:
    - Enrich property description only from ModelProperty not Model Type
    - Ignore to add specific discriminator property for child type
    - Set property name `"${propName}"` to handle special cases

### CODE REVIEW

File: packages/cadl-rlc-test/test/confidentialLedger/cadl-output/src/outputModels.ts [hunk 1]
```diff
@@ -61,7 +60,6 @@ export interface TransactionReceiptOutput {
   receipt: ReceiptContentsOutput;
   /** Possible values: Loading, Ready */
   state: string;
-  /** A unique identifier for the state of the ledger. If returned as part of a LedgerEntry, it indicates the state from which the entry was read. */
```
Comment (MaryGao) [inline packages/cadl-rlc-test/test/confidentialLedger/cadl-output/src/outputModels.ts:64]: These docs are tagged under the model type level not property level

File: packages/cadl-typescript/src/modelUtils.ts [hunk 2]
```diff
@@ -413,7 +413,12 @@ function getSchemaForModel(
     modelSchema.children?.immediate?.push(childSchema);
   }
 
-  if (discriminator) {
+  // Enable option `isPolyParent` and discriminator only when it has valid children
+  if (
+    discriminator &&
+    modelSchema?.children?.all?.length &&
+    modelSchema?.children?.all?.length > 0
+  ) {
```
Comment (MaryGao) [inline packages/cadl-typescript/src/modelUtils.ts:421]: Ignore to add specific discriminator property for child type

File: packages/cadl-typescript/src/modelUtils.ts [hunk 3]
```diff
@@ -427,14 +432,16 @@ function getSchemaForModel(
...
   }
 
   // applyExternalDocs(model, modelSchema);
   if (needRef) {
     return modelSchema;
   }
-  for (const [name, prop] of model.properties) {
+  for (const [propName, prop] of model.properties) {
+    const name = `"${propName}"`;
     if (!isSchemaProperty(program, prop)) {
```
Comment (MaryGao) [inline packages/cadl-typescript/src/modelUtils.ts:445]: Set name as `"${propName}"` to handle special cases

File: packages/cadl-typescript/src/transform/transformParameters.ts [hunk 4]
```diff
@@ -100,6 +100,10 @@ function getParameterMetadata(
 function getParameterName(name: string) {
   if (name === "content-type") {
```
Comment (qiaozha) [inline packages/cadl-typescript/src/transform/transformParameters.ts:101]: name.toLowerCase() === "content-type" ?

File: packages/cadl-typescript/test/integration/generated/models/propertyTypes/src/models.ts [hunk 5]
```diff
@@ -94,3 +94,9 @@ export interface DictionaryStringProperty {
   /** Property */
   property: Record<string, string>;
 }
+
+/** Model with a property never. (This property should not be included). */
+export interface NeverProperty {
+  /** Property */
+  property: never;
+}
```
Comment (qiaozha) [inline packages/cadl-typescript/test/integration/generated/models/propertyTypes/src/models.ts:102]: I think the proposal is to remove this never type property ?

Comment (MaryGao) [reply packages/cadl-typescript/test/integration/generated/models/propertyTypes/src/models.ts:102]: Will fix it in this issue: https://github.com/Azure/autorest.typescript/issues/1648

**Review by qiaozha — APPROVED** [review]

**Pull request merged**